content/chat/participant-activity.textile

```diff
@@ -1,7 +1,220 @@
+
+```[json]
+[
+  {
+    "channel":"channel0",
+    "presence":[
+      {"clientId": "user1", "action": "1"},
+      {"clientId": "user2", "action": "1"}
+    ]
+  },
+  {
+    "channel":"channel1",
+    "presence":[]
+  },
+  {
+    "channel":"channel2",
+    "presence":[
+      {"clientId": "user2", "action": "1"},
+      {"clientId": "user3", "action": "1"}
+    ]
+  }
+]
+```
+
+For more information making batch presence requests please refer to the "Presence documentation":/docs/presence-occupancy/presence#batch
+
+
+h3(#channel-separation). Channel separation
+
+Chat messages and chat participant status updates in one pub/sub channel would contain a large number of messages published and received. To avoid a potentially high volume of unnecessary messages, as well as the suggestion of only having your backend server being the exclusive subscriber to chat participants' online status updates, it is also advised to have separate pub/sub channels for separate purposes. One for a specific chat (e.g. general), while having a separate channel for the chat participants status (e.g. status-general) updates for that chat. For example, when issuing api keys to clients, assigning the clients the following "capabilities":https://ably.com/docs/auth/capabilities :
+
+```
+{
+  "status-general": ["presence"],
+  "general": ["publish", "history", "subscribe"],
+}
+```
+
+Meanwhile the server can also subscribe to the chat participants channel with an api key having the following capabilities:
+
+```
+{
+  "status-general": ["presence", "subscribe"],
+  "general": ["publish", "history", "subscribe"],
+}
+```
+
+h2(#last-seen). Last seen
+
+Along with determining whether a chat participant is actively online or not, another key piece of functionality in a chat application is to determine when the chat participant was last seen as online. So long as the client, the chat participant is using specifically, calls @enter@ on the presence set for that channel, you'll have a historical record of that entry and a record of when the chat participant left the presence set. Whether it was specifically calling @leave@ or they timed out due to an unstable connection.
+
+To retrieve the channel's presence history, call the channel's presence followed by its history function. Calling @history@ will retrieve up to 1000 records, starting from the latest presence entries. To find the @leave@ events for a specific chat participant, loop through each entry, first filtering out any records by the @clientId@ of your desired chat participant, and the action, which is @leave@. You then need to sort the records, finding the latest timestamp first. Below is an example of how you would implement this functionality:
+
+```[javascript]
+channel.attach(() => {
+  const presence = channel.presence;
```

Comment:
If we're only retrieving this from history then it'd make more sense to use REST, otherwise you'd already be part of the presence set, in which case you don't need to attach, as that's implicit.